repo.go

```diff
@@ -671,6 +672,38 @@ func (r *Repo) AddTargets(paths []string, custom json.RawMessage) error {
 	return r.AddTargetsWithExpires(paths, custom, data.DefaultExpires("targets"))
 }
 
+func (r *Repo) AddDigestTargets(digest string, length int64, custom json.RawMessage) error {
+	expires := data.DefaultExpires("targets")
+
+	t, err := r.targets()
+	if err != nil {
+		return err
+	}
+	hashes := make([]string, 1)
+	split_digest := strings.Split(digest, ":")
+	hashes[0] = split_digest[1]
+
+	meta := data.FileMeta{Length: length, Hashes: make(data.Hashes, len(hashes))}
+	meta.Hashes[split_digest[0]], err = hex.DecodeString(hashes[0])
+
+	path := digest
```

Comment:
Should the name of the target path be customizable? I can imagine use cases where a remote artifact name is associated with a hash, but not addressed directly hash.